### PR TITLE
Create batch job in gcp

### PR DESCRIPTION
**Pull request by destinychenlu**
example curl:

curl -X POST https://{launchpad_url}/v1/batch/example-job-202209210310 -H "Content-Type: application/json" -d '{"taskGroups":[{"taskSpec":{"runnables":[{"script":{"text":"echo Hello world! This is task ${BATCH_TASK_INDEX}. This job has a total of ${BATCH_TASK_COUNT} tasks."}}],"computeResource":{"cpuMilli":2000,"memoryMib":16},"maxRetryCount":2,"maxRunDuration":"3600s"},"taskCount":4,"parallelism":2}],"allocationPolicy":{"instances":[{"policy":{"machineType":"e2-standard-2"}}]},"labels":{"department":"finance","env":"testing"},"logsPolicy":{"destination":"CLOUD_LOGGING"}}'

### CODE REVIEW

File: src/main/java/com/snapchat/launchpad/batch/BatchController.java [hunk 1]
```diff
@@ -33,11 +35,16 @@ public ResponseEntity<String> relayBatchRequest(
...
-            @RequestBody(required = false) String rawBody)
-            throws IOException {
+            @RequestBody(required = false) String rawBody) {
         headers.remove(HttpHeaders.AUTHORIZATION);
-        return batchRelayer.relayRequestBatch(
-                path, HttpMethod.valueOf(request.getMethod()), params, headers, rawBody);
+        try {
+            batchRelayer.relayRequestBatch(
+                    path, HttpMethod.valueOf(request.getMethod()), params, headers, rawBody);
+            return ResponseEntity.ok().body("Successfully create batch job " + path);
```
Comment (wk93210) [inline src/main/java/com/snapchat/launchpad/batch/BatchController.java:43]: Can we return the response from batchRelayer.relayRequestBatch() instead? The original batch job response helps us debugging...

File: src/main/java/com/snapchat/launchpad/batch/utils/BatchRelayerGcp.java [hunk 2]
```diff
@@ -21,17 +30,29 @@ public ResponseEntity<String> relayRequestBatch(
...
+        String accessToken =
+                parseJson(MetadataConfig.getAttribute(AUTH_TOKEN_URI)).get(ACCESS_TOKEN).asText();
+        headers.add(HttpHeaders.AUTHORIZATION, String.format("Bearer %s", accessToken));
+
+        String zone = MetadataConfig.getZone();
         return relayRequest(
                 String.format(
-                        "batch.googleapis.com/v1/projects/%s/locations/%s/jobs",
-                        MetadataConfig.getProjectId(), MetadataConfig.getZone()),
+                        "https://batch.googleapis.com/v1/projects/%s/locations/%s/jobs?job_id=",
```
Comment (wk93210) [inline src/main/java/com/snapchat/launchpad/batch/utils/BatchRelayerGcp.java:41]: Just curious, what would the job_id be in this case? Does Google simply assign a random one?

**Review by wk93210 — APPROVED** [review]

File: src/main/java/com/snapchat/launchpad/batch/BatchController.java (hunk 1 above)
Comment (wk93210) [reply src/main/java/com/snapchat/launchpad/batch/BatchController.java:43]: Synced offline, can't return directly due to https://serverfault.com/questions/1070510/application-works-locally-but-fails-with-502-in-cloud-run-upstream-connect-e

Will extract body instead

File: src/main/java/com/snapchat/launchpad/batch/utils/BatchRelayerGcp.java (hunk 2 above)
Comment (destinychenlu) [reply src/main/java/com/snapchat/launchpad/batch/utils/BatchRelayerGcp.java:41]: We can construct job-id from snap side with timestamp. For example, example-job-20220921-partition-1. This is better for debugging if we want to look into any specific job

**Pull request merged**